nni/experiment/config/v1.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+
+from . import (
+    AlgorithmConfig,
+    CustomAlgorithmConfig,
+    ExperimentConfig,
+    KubeflowAzureStorageConfig,
+    KubeflowNfsConfig,
+    KubeflowRoleConfig,
+    RemoteMachineConfig,
+)
+from . import util
+
+def convert_to_v2(v1) -> ExperimentConfig:
+    platform = v1.pop('trainingServicePlatform')
+    if platform == 'pai':
+        platform = 'openpai'
+    v2 = ExperimentConfig(platform)
+
+    # see nni/tools/nnictl/config_schema.py for v1 field list
+
+    _drop_field(v1, 'authorName')
+    _move_field(v1, v2, 'experimentName', 'experiment_name')
+    _drop_field(v1, 'description')
+    _move_field(v1, v2, 'trialConcurrency', 'trial_concurrency')
+    _move_field(v1, v2, 'maxExecDuration', 'max_experiment_duration')
+    _move_field(v1, v2, 'maxTrialNum', 'max_trial_number')
+    v2.search_space_file = util.canonical_path(v1.pop('searchSpacePath', None))
+    _drop_field(v1, 'multiPhase')
+    _drop_field(v1, 'multiThread')  # FIXME
+    _move_field(v1, v2, 'nniManagerIp', 'nni_manager_ip')
+    _move_field(v1, v2, 'logDir', 'experiment_working_directory')
+    _move_field(v1, v2, 'debug', 'debug')
+    _drop_field(v1, 'versionCheck')
+    _move_field(v1, v2, 'logLevel', 'log_level')
+    _drop_field(v1, 'logCollection')
+    _drop_field(v1, 'useAnnotation')  # FIXME
```

Comment:
`useAnnotation` is handled in nnictl with some hacks. I think it's not elegant but don't want to solve it in this release.
`multiThread` is not supported here. It's easy to implement since the value is directly passed to dispatcher, but I don't want to expose this interface in v2 schema.